nifi-nar-bundles/nifi-media-bundle/nifi-media-processors/src/main/java/org/apache/nifi/processors/media/ExtractMediaMetadata.java

```diff
@@ -0,0 +1,276 @@
+@WritesAttributes({@WritesAttribute(attribute = "tika.<attribute>", description = "The extracted content metadata "
+        + "will be inserted with the attribute name \"tika.<attribute>\". ")})
+@SupportsBatching
+public class ExtractMediaMetadata extends AbstractProcessor {
+
+    public static final PropertyDescriptor MAX_NUMBER_OF_ATTRIBUTES = new PropertyDescriptor.Builder()
+            .name("Max Number of Attributes")
+            .description("Specify the max number of attributes to add to the flowfile. There is no guarantee in what order"
+                    + " the tags will be processed. By default it will process all of them.")
+            .required(false)
+            .addValidator(StandardValidators.NON_NEGATIVE_INTEGER_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor FILENAME_FILTER = new PropertyDescriptor.Builder()
+            .name("File Name Filter")
+            .description("A regular expression identifying file names which metadata should extracted.  As flowfiles"
+                    + " are processed, if the file name matches this regular expression or this expression is"
+                    + " blank, the flowfile will be scanned for it's MIME type and metadata.  If left blank, all"
+                    + " flowfiles will be scanned.")
+            .required(false)
+            .addValidator(StandardValidators.REGULAR_EXPRESSION_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor MIME_TYPE_FILTER = new PropertyDescriptor.Builder()
+            .name("MIME Type Filter")
+            .description("A regular expression identifying MIME types for which metadata should extracted.  Flowfiles"
+                    + " selected for scanning by the File Name Filter are parsed to determine the MIME type and extract"
+                    + " metadata.  If the MIME type found matches this regular expression or this expression is"
+                    + " blank, the metadata keys that match the Metadata Key Filter will be added to the flowfile"
+                    + " attributes.  There is no guarantee in what order attributes will be produced.  If"
+                    + " left blank, metadata will be extracted from all flow files selected for scanning.")
+            .required(false)
+            .addValidator(StandardValidators.REGULAR_EXPRESSION_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor METADATA_KEY_FILTER = new PropertyDescriptor.Builder()
+            .name("Metadata Key Filter")
+            .description("A regular expression identifying which metadata keys received from the parser should be"
+                    + " added to the flowfile attributes.  If left blank, all metadata keys parsed will be added to the"
+                    + " flowfile attributes.")
+            .required(false)
+            .addValidator(StandardValidators.REGULAR_EXPRESSION_VALIDATOR)
+            .build();
```

Comment:
Should support for expression language be added here? As an example, in the event multiple different types of data are coming through one processor the user may want to construct the regex using attributes on the flowfile